### PR TITLE
397 correct epidata test fail: correction for Eisenach not anymore in county table

### LINKED ISSUE #397: Epidata tests fail
The epidata tests fail since the official document of destatis was changed and Eisenach (16056) is no longer in the list.

The failing test is:

```def test_get_nuts3_county_id_map(self, mock_print)```

but not all implications are clear to me. We should check what this means for all data frames.

### PR DESCRIPTION
**Pull request by mknaranja**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [x] New code adheres with the [coding guidelines](coding-guidelines)
* [x] Tests for new functionality has been added
* [x] A local test was succesful
* [ ] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [x] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [x] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [x] There are appropriate **unit tests** and they pass.
* [x] The git history is clean and linearized for the merge request.
* [x] Coverage report for new code is acceptable. 

Closes #397

### CODE REVIEW

**Comment by mknaranja** [comment]
Hope that it now goes through. Please have a look at this soon @patricklnz

File: pycode/memilio-epidata/memilio/epidata_test/test_epidata_geoModificationGermany.py [hunk 1]
```diff
@@ -546,8 +542,10 @@ def test_merge_df_counties(self):
             test_df, 16063, [16063, 16056], group_columns, group_columns)
         pd.testing.assert_frame_equal(result_df, self.eisenach_merged_df)
         # the test dataframe should be unchanged as it is the input of the function
+        # avoid error due to comparison of int32 and int64 columns.
```
Comment (patricklnz) [inline pycode/memilio-epidata/memilio/epidata_test/test_epidata_geoModificationGermany.py:545]: Where does this error come from?

File: pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_population_data.py [hunk 2]
```diff
@@ -134,8 +134,10 @@ def test_get_population(self, mock_data):
             read_data=True, file_format='json', out_folder=self.path,
             no_raw=False, split_gender=False, merge_eisenach=False)
 
+        # add seven to the number of test_counties as this is the current workaround to add counties which are
+        # not mentioned in old data sets
         test_df = pd.read_json(os.path.join(
-            self.path, 'Germany/', 'county_current_population_dim401.json'))
+            self.path, 'Germany/', 'county_current_population_dim' + str(len(self.test_counties)+7) +'.json'))
```
Comment (patricklnz) [inline pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_population_data.py:140]: I think the function `get_new_counties` should not add those 7 counties if there is no data for them.
The workaround should be okay for now, but maybe we should open a new Issue or at least a TODO.

File: pycode/memilio-epidata/memilio/epidata/getPopulationData.py [hunk 3]
```diff
@@ -535,8 +532,8 @@ def get_population_data(read_data=dd.defaultDict['read_data'],
             filename = 'county_current_population'
             filename_raw = 'county_population'
         else:  # Write Dataframe without merging
-            filename = 'county_current_population_dim401'
-            filename_raw = 'county_population_dim401'
+            filename = 'county_current_population_dim' + str(len(df_current))
+            filename_raw = 'county_population_dim' + str(len(df_current))
```
Comment (patricklnz) [inline pycode/memilio-epidata/memilio/epidata/getPopulationData.py:536]: Maybe print a warning if dimension is not 400 or 401 (which should now not be possible)

File: pycode/memilio-epidata/memilio/epidata/getPopulationData.py [hunk 4]
```diff
@@ -523,7 +520,7 @@ def get_population_data(read_data=dd.defaultDict['read_data'],
         directory = os.path.join(out_folder, 'Germany/')
         gd.check_dir(directory)
 
-        if merge_eisenach == True:
+        if len(df_current) == 400 or merge_eisenach == True:
```
Comment (patricklnz) [inline pycode/memilio-epidata/memilio/epidata/getPopulationData.py:523]: `merge_eisenach` can be removed.

**Review by patricklnz — CHANGES_REQUESTED** [review]
Minor adjustments. Else looks good to me.

File: pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_population_data.py (hunk 2 above)
Comment (mknaranja) [reply pycode/memilio-epidata/memilio/epidata_test/test_epidata_get_population_data.py:140]: See https://github.com/DLR-SC/memilio/issues/405

File: pycode/memilio-epidata/memilio/epidata/getPopulationData.py [hunk 5]
```diff
@@ -136,7 +136,7 @@ def load_population_data(out_folder=dd.defaultDict['out_folder'],
    @param read_data False or True. Defines if data is read from file or downloaded. Default defined in defaultDict.
    @param no_raw True or False. Defines if unchanged raw data is written or not. Default defined in defaultDict.
    @param file_format File format which is used for writing the data. Default defined in defaultDict.
-   @return 3 Dataframes of migration, reg_key and zensus
+   @return Three dataframes of county_table, reg_key and zensus data.
```
Comment (patricklnz) [inline pycode/memilio-epidata/memilio/epidata/getPopulationData.py:139]: Line 123:
`"""! Load of counties, zensus and reg_key files` 
should be:
`"""! Load of county_table, zensus and reg_key files`.

Line 127:
`data for population per county [stored in "county_table"]`

**Review by patricklnz — CHANGES_REQUESTED** [review]
Last change.

**Review by patricklnz — APPROVED** [review]
Approved.

**Pull request merged**